server.js

```diff
@@ -20,4 +20,57 @@ app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
+app.get('/messages', function (request, response) {
+  response.json(messages)
+})
+
+app.post('/messages', function (request, response) {
+  let newMessage = {};
+
+  if (request.body.from.trim() && request.body.text.trim()) {
+    newMessage.id = messages.length
+    newMessage.from = request.body.from
+    newMessage.text = request.body.text
+    newMessage.timeSent = new Date()
+    messages.push(newMessage)
+    response.json(messages)
+  } else {
+    response.status(400).json("Please fill all fileds")
+  }
+
+})
+
+app.get('/messages/search', function (request, response) {
+  let searchWord = request.query.text
+  let result = messages.filter(item => item.text.toLowerCase().includes(searchWord.toLowerCase()))
```

Comment:
It would be more efficient to set word.toLowerCase() to a new variable upfront, and pass this into the filter method.

Reply:
```suggestion
  let searchWord = request.query.text.toLowerCase()
  let result = messages.filter(item => item.text.toLowerCase().includes(searchWord))
```
is this what you mean?